plutus-tx-plugin/src/PlutusTx/Compiler/Expr.hs

```diff
@@ -848,6 +857,280 @@ compileExpr e = withContextM 2 (sdToTxt $ "Compiling expr:" GHC.<+> GHC.ppr e) $
+                [GHC.Type ty, _ordDict, GHC.Lit (GHC.LitNumber _ i), GHC.Lit (GHC.LitNumber _ j)]
+                    | ty `GHC.eqType` GHC.integerTy -> Just $ do
+                        res <- lookupDataCon =<< thNameToGhcNameOrFail (if i > j then 'True else 'False)
+                        compileDataConRef res
+                _ -> Nothing
+            )
+          , ( '(GHC.Classes.>=)
+            , \case
+                [GHC.Type ty, _ordDict]
+                    | ty `GHC.eqType` GHC.integerTy ->
+                        Just $
+                            compileExpr . GHC.Var =<< lookupId =<< thNameToGhcNameOrFail 'Builtins.greaterThanEqualsInteger
+                [GHC.Type ty, _ordDict, GHC.Lit (GHC.LitNumber _ i), GHC.Lit (GHC.LitNumber _ j)]
+                    | ty `GHC.eqType` GHC.integerTy -> Just $ do
+                        res <- lookupDataCon =<< thNameToGhcNameOrFail (if i >= j then 'True else 'False)
+                        compileDataConRef res
+                _ -> Nothing
+            )
+          , ( '(GHC.Num.+)
+            , \case
+                [GHC.Type ty, _numDict]
+                    | ty `GHC.eqType` GHC.integerTy ->
+                        Just $
+                            compileExpr . GHC.Var =<< lookupId =<< thNameToGhcNameOrFail 'BI.addInteger
+                [GHC.Type ty, _numDict, GHC.Lit (GHC.LitNumber numTy i), GHC.Lit (GHC.LitNumber _numTy j)]
+                    | ty `GHC.eqType` GHC.integerTy ->
+                        Just . compileExpr . GHC.Lit $ GHC.LitNumber numTy (i + j)
+                _ -> Nothing
+            )
+          , ( '(GHC.Num.-)
+            , \case
+                [GHC.Type ty, _numDict]
+                    | ty `GHC.eqType` GHC.integerTy ->
+                        Just $
+                            compileExpr . GHC.Var =<< lookupId =<< thNameToGhcNameOrFail 'BI.subtractInteger
+                [GHC.Type ty, _numDict, GHC.Lit (GHC.LitNumber numTy i), GHC.Lit (GHC.LitNumber _numTy j)]
+                    | ty `GHC.eqType` GHC.integerTy ->
+                        Just . compileExpr . GHC.Lit $ GHC.LitNumber numTy (i - j)
+                _ -> Nothing
+            )
+          , ( '(GHC.Num.*)
+            , \case
+                [GHC.Type ty, _numDict]
+                    | ty `GHC.eqType` GHC.integerTy ->
+                        Just $
+                            compileExpr . GHC.Var =<< lookupId =<< thNameToGhcNameOrFail 'BI.multiplyInteger
+                [GHC.Type ty, _numDict, GHC.Lit (GHC.LitNumber numTy i), GHC.Lit (GHC.LitNumber _numTy j)]
+                    | ty `GHC.eqType` GHC.integerTy ->
+                        Just . compileExpr . GHC.Lit $ GHC.LitNumber numTy (i * j)
+                _ -> Nothing
+            )
+          , ( 'GHC.Num.fromInteger
+            , \case
+                -- Compile `fromInteger @Integer` via `id`.
```

Comment:
Does this actually come up much? Also does the base version not work given that it's presumably implemented as `id`?